Statistic data

MrZhenShen opened the pull request:
# Description
## Change type: #minor

Statistic data [RIP-40](https://sigma-internship-project.atlassian.net/browse/RIP-40?atlOrigin=eyJpIjoiMWQxNTNmMzAxYjBkNDQ2YTkzMGRkZGM0OWQzMjIyZjAiLCJwIjoiaiJ9)

* Create a scheduler to retrieve the latest data from external API
* Create endpoints:
1.  Get the latest statistical data
2. Get data to create a histogram of specific losses for the whole time of war
3. Get math calculations (max, min, mean, median)
* Implement API testing
* Create: WarAPIService, ScheduledStatisticDataRetriever, StatisticDataController, StatisticDataControllerTest, LossDayDto, CalculatedStatisticDataDto

MrZhenShen committed:
Commit to review
Implement scheduler
Add staging dtos for StatisticDataDto

MrZhenShen committed:
Make warDate, dayNumber unique
Clear printlns
Change sort type in StatisticDataRepository

MrZhenShen committed:
Add endpoint Request GET a map of losses with its date

MrZhenShen committed:
Add endpoints to GET min, max, mean and median of losses by type

MrZhenShen committed:
Change http status exception if result of findAll is empty

MrZhenShen committed:
Merge branch 'main' into statistic-data

MrZhenShen committed:
Merge branch 'main' into statistic-data

# Conflicts:
#	pom.xml

MrZhenShen committed:
Delete non-used path in was API service
Make math endpoints in one
Replace web exceptions with empty values

eduard-romaniuk committed:
Merge branch 'main' into statistic-data

MrZhenShen committed:
Merge remote-tracking branch 'origin/statistic-data' into statistic-data

MrZhenShen committed:
Create SchedulerConfigProperties for cron expressions
Delete 404 response for latest data endpoint

MrZhenShen committed:
Remove configurations for cron in scheduler

MrZhenShen committed:
Remove exception from nullable response in the latest data endpoint
Create sql tasks for statistic_data testing
Create StatisticDataControllerTest

eduard-romaniuk reviewed (changes requested):
I have a bit more comments to tests, but let's first fix comments to controller and assertions

MrZhenShen committed:
Create for responses CalculatedStatisticDataDto, LossDayDto
Change response types for /dataset and /math
Merge math methods in one
Simplify unit tests

MrZhenShen committed:
Add Internal Server Error Exception if repository returns no data

MrZhenShen committed:
Merge branch 'main' into statistic-data

# Conflicts:
#	pom.xml

MrZhenShen committed:
Permit /statistic-data/latest for everyone.
Create tests with different auth user permissions
Make permissions for /statistic-data endpoints
